scripts/spack_packages/umpire/package.py

```diff
@@ -118,290 +83,149 @@ class Umpire(CMakePackage, CudaPackage):
-        else:
-            cfg.write(cmake_cache_option("ENABLE_DEVELOPER_BENCHMARKS", False))
-            cfg.write(cmake_cache_option("ENABLE_BENCHMARKS", False))
-
+        entries = [x for x in entries if not 'COMPILER_ID' in x]
+        return entries
 
-        if "+ipc_shmem" in spec:
-            cfg.write(cmake_cache_option("ENABLE_IPC_SHARED_MEMORY", True))
-        else:
-            cfg.write(cmake_cache_option("ENABLE_IPC_SHARED_MEMORY", False))
 
+    def initconfig_hardware_entries(self):
+        spec = self.spec
+        entries = super(Umpire, self).initconfig_hardware_entries()
 
+        entries.append(cmake_cache_option("ENABLE_CUDA", "+cuda" in spec))
         if "+cuda" in spec:
-            cfg.write("#------------------{0}\n".format("-" * 60))
-            cfg.write("# Cuda\n")
-            cfg.write("#------------------{0}\n\n".format("-" * 60))
-
-            cfg.write(cmake_cache_option("ENABLE_CUDA", True))
-
-            cudatoolkitdir = spec['cuda'].prefix
-            cfg.write(cmake_cache_entry("CUDA_TOOLKIT_ROOT_DIR",
-                                        cudatoolkitdir))
-            cudacompiler = "${CUDA_TOOLKIT_ROOT_DIR}/bin/nvcc"
-            cfg.write(cmake_cache_entry("CMAKE_CUDA_COMPILER",
-                                        cudacompiler))
-
             cuda_flags = []
-
             if not spec.satisfies('cuda_arch=none'):
                 cuda_arch = spec.variants['cuda_arch'].value
                 cuda_flags.append('-arch sm_{0}'.format(cuda_arch[0]))
 
-            if '+deviceconst' in spec:
-                cfg.write(cmake_cache_option("ENABLE_DEVICE_CONST", True))
-
-            if using_toolchain:
-                cuda_flags.append("-Xcompiler {}".format(using_toolchain[0]))
-
-            cfg.write(cmake_cache_string("CMAKE_CUDA_FLAGS",  ' '.join(cuda_flags)))
+            if self.spec_uses_toolchain(self.spec):
+                cuda_flags.append("-Xcompiler {}".format(self.spec_uses_toolchain(self.spec)[0]))
 
-        else:
-            cfg.write(cmake_cache_option("ENABLE_CUDA", False))
+            entries.append(cmake_cache_string("CMAKE_CUDA_FLAGS",  ' '.join(cuda_flags)))
 
+        entries.append(cmake_cache_option("ENABLE_HIP", "+hip" in spec))
         if "+hip" in spec:
-            cfg.write("#------------------{0}\n".format("-" * 60))
-            cfg.write("# HIP\n")
-            cfg.write("#------------------{0}\n\n".format("-" * 60))
-
-            cfg.write(cmake_cache_option("ENABLE_HIP", True))
-
-#            -DHIP_ROOT_DIR=/opt/rocm-3.6.0/hip -DHIP_CLANG_PATH=/opt/rocm-3.6.0/llvm/bin
-
             hip_root = spec['hip'].prefix
             rocm_root = hip_root + "/.."
-            cfg.write(cmake_cache_entry("HIP_ROOT_DIR",
+            entries.append(cmake_cache_string("HIP_ROOT_DIR",
                                         hip_root))
-            cfg.write(cmake_cache_entry("HIP_CLANG_PATH",
+            entries.append(cmake_cache_string("HIP_CLANG_PATH",
```

Comment:
Is there a cmake_cache_entry to treat the variable as a PATH? I won't change the behavior, I think, but... well it seems more appropriate.